Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Db.ts

```diff
@@ -134,3 +127,30 @@ export const asUpsert = (session: ClientSession): ReplaceOptions => ({ upsert: t
 
 // MongoDB FindOption to return at most 5 documents
 export const limitFive = (session: ClientSession): FindOptions => ({ limit: 5, session });
+
+/**
+ * Lock in-use meadowlark documents, both those being directly updated and those being referenced.
+ */
+export async function lockDocuments(
+  concurrencyCollection: Collection<ConcurrencyDocument>,
+  concurrencyDocuments: ConcurrencyDocument[],
+  session: ClientSession,
+): Promise<void> {
+  await concurrencyCollection.insertMany(concurrencyDocuments, { session });
+
+  // eslint-disable-next-line no-underscore-dangle
+  const documentUuids: DocumentUuid[] = concurrencyDocuments.map((document) => document._id);
+
+  await concurrencyCollection.deleteMany(
+    {
+      $and: [
```

Comment:
Are these `$and`s necessary?

Reply:
Good point. Now that we have just `documentUuid`, this and is not necessary.